### PR TITLE
[POAE7-2373] Fix Float and GroupBy AVG with null value

### PR DESCRIPTION
**Pull request by Deegue**
### What changes were proposed in this pull request?
Float support in AVG and some parts of GROUP BY.
Enable several test cases of cider-velox.

### Why are the changes needed?
Bug fix.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
UTs.

### Which label does this PR belong to?
BUG

### CODE REVIEW

**Comment by Deegue** [comment]
Just found float is fixed in [[POAE7-2370]](https://github.com/intel/BDTK/pull/41/files).

File: cider/exec/module/CiderRuntimeModule.cpp [hunk 1]
```diff
@@ -820,15 +821,35 @@ CiderBatch CiderRuntimeModule::setSchemaAndUpdateAggResIfNeed(
...
+        } else if (target_type == SQLTypes::kFLOAT) {
+          float* result = const_cast<float*>(
+              reinterpret_cast<const float*>(outBuffers[flatten_index]));
+          double* cast_buffer = reinterpret_cast<double*>(result);
+          if (result[0] == kFloatNullValue) {
+            cast_buffer[0] = kDoubleNullValue;
+          } else {
+            cast_buffer[0] = (double)result[0];
+          }
+        } else if (target_type == SQLTypes::kDOUBLE) {
```
Comment (yma11) [inline cider/exec/module/CiderRuntimeModule.cpp:844]: It's about the sum(double)->double case, right? if so, this case in L848 won't happen in cider.

File: cider/exec/operator/aggregate/CiderAggTargetColExtractorTmpl.h [hunk 2]
```diff
@@ -161,7 +161,11 @@ class AVGAggExtractor : public CiderAggTargetColExtractor {
 
       AVGT sumVal = *reinterpret_cast<const SUMT*>(rowPtr + sum_offset_);
       int64_t countVal = *reinterpret_cast<const COUNTT*>(rowPtr + count_offset_);
-      avgResultVector[i] = sumVal / countVal;
+      if (sumVal == 0 && countVal == 0) {
```
Comment (yma11) [inline cider/exec/operator/aggregate/CiderAggTargetColExtractorTmpl.h:164]: maybe we don't need check `sumVal`?

**Comment by Deegue** [comment]
Working in progress to support partial avg in `SimpleAggExtractor`.

File: cider/exec/module/CiderRuntimeModule.cpp [hunk 3]
```diff
@@ -742,10 +742,16 @@ void CiderRuntimeModule::initCiderAggTargetColExtractors() {
         CiderAggTargetColExtractorBuilder::buildCiderAggTargetColExtractor(
             group_by_agg_hashtable_.get(), target_col_index, is_partial_avg_sum[i]);
 
-    // TODO: Partial Avg
     if (kAVG == col_info.agg_type) {
```
Comment (yma11) [inline cider/exec/module/CiderRuntimeModule.cpp:745]: For partial AVG, the `col_info.agg_type` should still be `kSUM `and `kCount`?

Comment (Deegue) [reply cider/exec/module/CiderRuntimeModule.cpp:745]: No, it's not correct, I think I misunderstood before. Still working on another PR and added WIP label..

**Review by yma11 — DISMISSED** [review]

**Comment by yma11** [comment]
LGTM now, let's do the test before merge.

**Comment by Deegue** [comment]
> should we merge this?

we have to merge https://github.com/Intel-bigdata/velox/pull/10 and retest this pr before it's merged.

**Review by yma11 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**